vmdb/app/controllers/ems_infra_controller.rb

```diff
@@ -33,41 +33,56 @@ def scaling
       return
     end
 
-    @count_parameters = @stack.parameters.select { |x| x.name.include?('::count') }
+    @orig_parameters = @stack.parameters.select { |x| x.name.include?('::count') }
 
     return unless params[:scale]
 
-    scale_parameters = params.select { |k, _v| k.include?('::count') }
-    assigned_hosts = scale_parameters.values.sum(&:to_i)
+    form_parameters = params.select { |k, _v| k.include?('::count') }
+
+    validate_selected_hosts_not_exceed_available_hosts(form_parameters)
+
+    changed_parameters = []
+    return_message = _("Scaling")
+
+    select_changed_parameters_and_return_message(form_parameters, changed_parameters, return_message)
+    validate_a_change_was_made(changed_parameters)
+
+    return unless changed_parameters.length > 0
+
+    begin
+      @stack.raw_update_stack(:parameters => changed_parameters)
+      redirect_to :action => 'show', :id => params[:id], :flash_msg => return_message
+    rescue => ex
+      log_and_flash_message(_("Unable to initiate scaling: %s") % ex)
+    end
+  end
+
+  private ############################
+
+  def select_changed_parameters_and_return_message(form_parameters, changed_parameters, return_message)
```

Comment:
when you change state or param inside of the method, it's better to have ! in the end, so:

select_changed_parameters_and_return_message!

Though I prefer not to change the params and rather return them in most cases, you can return multiple variables like in python. But If other guys are ok with this, it's not that much of a problem